### PR TITLE
Run3-gex53 Correct ECAL algorithms for dd4hep

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Correct ECAL algorithms for dd4hep (treatment of units)

#### PR validation:

Tested using cfg's in SimG4Core/Configuration/test

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/EcalCommonData/plugins/dd4hep/DDEcalBarrelNewAlgo.cc [hunk 1]
```diff
@@ -1146,7 +1150,7 @@ static long algorithm(dd4hep::Detector& /* description */, cms::DDParsingContext
             }
             unsigned int femCopy(0);
             for (unsigned int ilyFEM(0); ilyFEM != ily.vecIlyFEMZ.size(); ++ilyFEM) {
-              const double radius(ilyRMax - 1_mm - ily.fEMHeightHalf);
+              const double radius(ilyRMax - 1 * dd4hep::mm - ily.fEMHeightHalf);
```
Comment (civanch) [inline Geometry/EcalCommonData/plugins/dd4hep/DDEcalBarrelNewAlgo.cc:1153]: @bsunanda , why we need "1 * dd4hep::mm"? May it be "dd4hep::mm"? or compiler is cleaver enough to recognize this?

File: Geometry/EcalCommonData/plugins/dd4hep/DDEcalBarrelNewAlgo.cc [hunk 2]
```diff
@@ -1440,8 +1445,8 @@ static long algorithm(dd4hep::Detector& /* description */, cms::DDParsingContext
 
     // theta is angle in yz plane between z axis & leading edge of crystal
     double theta(90_deg);
-    double zee(0_mm);
-    double side(0_mm);
+    double zee(0 * dd4hep::mm);
```
Comment (civanch) [inline Geometry/EcalCommonData/plugins/dd4hep/DDEcalBarrelNewAlgo.cc:1448]: why not simply (0.) in all places?

**Comment by civanch** [comment]
bug

**Comment by civanch** [comment]
bug-fix

**Comment by bsunanda** [comment]
@cvuosalo @civanch specification of unit says the type of measurement. I can remove that for 0 but for 1mm it is better to be explicit. I am in 2 mind what is the best even for 0.

**Comment by cvuosalo** [comment]
I prefer `1 * dd4hep::mm`. For zero, I would prefer just `double zee(0.);` since zero is units-independent.

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/EcalCommonData/plugins/dd4hep/DDEcalBarrelNewAlgo.cc (hunk 2 above)
Comment (bsunanda) [reply Geometry/EcalCommonData/plugins/dd4hep/DDEcalBarrelNewAlgo.cc]: using 0 now

File: Geometry/EcalCommonData/plugins/dd4hep/DDEcalBarrelNewAlgo.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/EcalCommonData/plugins/dd4hep/DDEcalBarrelNewAlgo.cc:1149]: keeping 1 to make it look 1 mm

**Comment by civanch** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**